corehq/apps/linked_domain/static/linked_domain/js/domain_links.js

```diff
@@ -166,10 +167,58 @@ hqDefine("linked_domain/js/domain_links", [
         return self;
     };
 
-    var AddDownstreamDomainModal = function (manageDomainsViewModel, availableDomains) {
+    var PullReleaseContentViewModel = function (data) {
+        // Pull Content Tab
+        self.parent = data.parent;
+        self.linkedDataViewModels = data.linkedDataViewModels;
+        self.upstreamLink = data.upstreamLink;
+        if (self.upstreamLink) {
+            if (self.upstreamLink.is_remote) {
```

Comment:
might be good to comment on what's going on here at a high level. Not 100% sure what's happening?

Reply:
Instead of adding a comment, I refactored this to make it more straightforward. The django side sets an `upstream_url` and `downstream_url` on the domain link view model that the frontend can use directly. I also added tests around creating urls from domain links for both remote and non-remote links.